src/frontend/web_application/src/scenes/ContactBook/presenter.jsx

```diff
@@ -62,17 +59,14 @@ class ContactBook extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      tags: [],
+      tags: [].concat(generateStateFromProps.bind(props)),
```

Comment:
nope :)

you might not have contacts at this time.
Also I think we should not define calculated values into a constructor: just define the structure of the state, in other word: always use setState.

In our convention, `generateStateFromProps` is used in componentWillMount (when props are ready) and `componentWillReceiveProps` (c.f ReplyForm Component)